warehouse/accounts/models.py

```diff
@@ -111,6 +120,22 @@ class UnverifyReasons(enum.Enum):
     SoftBounce = "soft bounce"
 
 
+class TwoFactor(db.ModelBase):
```

Comment:
Is there a reason to have a second table (with ``lazy=False``, so a join will be generated by default in every query) and an index to store a single 20 byte binary? It seems like it should be just as fine to add the ``totp_secret`` column to the ``UserModel`` itself. The only reason I can think of for making it a separate table is if we plan on allowing multiple TOTP secrets to be registered per user, but currently there is a unique constraint preventing that (and the code base is not setup to handle multiple).

Reply:
The idea here was to provide breathing space for the additional column(s) that U2F/Webauthn will require (and therefore to minimize the number of `User` migrations`).

In retrospect, though, that's a little shaky: there's no real semantic connection between TOTP and U2F/Webauthn data, and having them share the same table would make configurations like "one TOTP, many security keys" difficult to model. In light of that, we could drop this back down to a single column on the `User` model and add a separate `SecurityKey` model a la `Email` when the time comes. Thoughts?